packages/runtime/src/types.ts

```diff
@@ -166,3 +172,10 @@ export type ZodSchemas = {
      */
     input?: Record<string, Record<string, z.ZodSchema>>;
 };
+
+export type CustomEncryption = {
+    encrypt: (model: string, field: FieldInfo, plain: string) => Promise<string>;
+    decrypt: (model: string, field: FieldInfo, cipher: string) => Promise<string>;
+};
+
+export type SimpleEncryption = { encryptionKey: string };
```

Comment:
Since we don't need to represent the key in plain text anywhere, is it better to define the key as a UInt8Array instead? So that we don't need to imply any encoding form or limit it to be a utf-8 string.

Reply:
If we do so, we can also validate if the given key is 32-byte long.

Reply:
I think we still need to keep it a `string` since the user would actually provide it as a `string`.

Maybe keep it as string, but ensure it is a valid `UInt8Array`?

@ymc9  what do you think?

Reply:
I'm think the user will use it like:

```ts
const getEncryptionKey() {
  return new Uint8Array(Buffer.from(process.env.ENCRYPTION_KEY, 'base64'));
}

const db = enhance(prisma, { user }, { encryption: { encryptionKey: getEncryptionKey() } });
```

He can also store the key as binary form somewhere and then load it directly and pass to the `enhance` call. My main thought is that zenstack can be agnostic about the key storage form.

Reply:
Would they be able to do simply:

`const db = enhance(prisma, { user }, { encryption: { encryptionKey: "c558Gq0YQK2QcqtkMF9BGXHCQn4dMF8w" } });`

With the field-encryption extension, it allows passing in a key as a string (I believe it validates its length at runtime)

Reply:
Yes, I see that field-encryption uses https://github.com/47ng/cloak to manage encryption keys - which seems to use a special encoding form (base64 + some annotations).

We can improve the options to allow base64 strings as well in the future.